source/common/grpc/google_grpc_creds_impl.cc

```diff
@@ -0,0 +1,78 @@
+#include "common/grpc/google_grpc_creds_impl.h"
+
+#include "envoy/api/v2/core/grpc_service.pb.h"
+#include "envoy/grpc/google_grpc_creds.h"
+#include "envoy/registry/registry.h"
+
+#include "common/config/datasource.h"
+
+namespace Envoy {
+namespace Grpc {
+
+/**
+ * TODO: Create GoogleGrpcCredentialsFactory for each built-in credential type defined in Google
+ * gRPC.
+ */
+
+const grpc::SslCredentialsOptions BuildSslOptionsFromConfig(
```

Comment:
The `const` bit isn't really doing anything. Also, nit, should be a `buildSsl...`